test/integration/convert_partial_specialization_test.cpp

```diff
@@ -0,0 +1,89 @@
+#include "yaml-cpp/emitterstyle.h"
+#include "yaml-cpp/eventhandler.h"
+#include "yaml-cpp/yaml.h"  // IWYU pragma: keep
+#include "gtest/gtest.h"
+
+// Base class
+class A {
+ public:
+  // virtual load/emit methods
+  virtual void load(const YAML::Node &node) { a = node["a"].as<int>(); }
+
+  virtual YAML::Node emit() const {
+    YAML::Node node;
+    node["a"] = a;
+    return node;
+  }
+
+  int a;
+};
+
+// Derived class
+class B : public A {
+ public:
+  // override virtual load/emit methods
+  virtual void load(const YAML::Node &node) override {
+    A::load(node);
+    b = node["b"].as<int>();
+  }
+
+  virtual YAML::Node emit() const override {
+    YAML::Node node = A::emit();
+    node["b"] = b;
+    return node;
+  }
+
+  int b;
+};
+
+// Implementation of convert::{encode,decode} for all classes derived from or
+// being A
+namespace YAML {
+template <typename T>
+struct convert<T, typename std::enable_if<std::is_base_of<A, T>::value>::type> {
+  static Node encode(const T &rhs) {
+    Node node = rhs.emit();
+    return node;
+  }
+
+  static bool decode(const Node &node, T &rhs) {
+    rhs.load(node);
+    return true;
+  }
+};
+
+namespace {
+
+TEST(ConvertPartialSpecializationTest, encode_base) {
```

Comment:
`encode_base -> EncodeBaseClass` just like the other test cases' name.

Reply:
Done